app/controllers/works_controller.rb

```diff
@@ -0,0 +1,86 @@
+class WorksController < ApplicationController
+  # before_action :get_work , only: [:show, :edit, :update, :destroy]
+
+  # def get_work
+  #   @work = Work.find(params[:id])
+  # end
+
+  def index
+    @works = Work.all.order(:category, :title)
+  end
+
+  def show
+    # @work = Work.find(params[:id])
+    find_work_by_params_id
+  end
+
+  def new
+    @work = Work.new
+  end
+
+  def create
+    @work = Work.new(work_params)
+
+    if @work.save
+      flash[:status] = :success
+      flash[:message] = "Successfully created work #{@work.id}"
+      redirect_to works_path
+      return
+    else
+      flash.now[:status] = :failure
+      flash.now[:message] = "Failed to create new work"
+      flash.now[:details] = @work.errors.messages
+      render :new, status: :bad_request
+    end
+  end
+
+  def edit
+    # @work = Work.find(params[:id])
+    find_work_by_params_id
+  end
+
+  def update
+    # @work = Work.find(params[:id])
+    if find_work_by_params_id
+      @work.update_attributes(work_params)
+      if @work.save
+        redirect_to work_path(@work)
+        return
```

Comment:
If updating the work fails, this will not report the reasons why back to the user. Displaying this information is important to help your users understand what went wrong, but it is also an important tool for you the developer so that you can diagnose when things go wrong.